File: Data Structures & Algorithms/products-of-array-discluding-self/submission-7.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        # len(nums) = len(nums)
        a, b = [None] * len(nums), [None] * len(nums)
        a[0] = nums[0]
        b[len(nums) - 1] = nums[-1]
        for i in range(1, len(nums)):
            a[i] = nums[i] * a[i - 1]
            b[len(nums) - i - 1] = nums[len(nums) - 1 - i] * b[len(nums) - i]

        ans = [None] * len(nums)
        ans[0] = b[1]
        ans[len(nums) - 1] = a[-2]
        for i in range(1, len(nums) - 1):
            ans[i] = (a[i-1] * b[i+1])
        return ans

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-7.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const obj = new Set(nums);
        nums = nums.map(x => {
            if (!obj.has(x - 1)) return x;
        }).filter(x => x!== undefined);

        nums = nums.map(x => {
            let h = 0;
            while(obj.has(x)) {
                x++;
                h++
            };
            return h;
        });
        return Math.max(...nums, 0);
    }
}
